From d54ac9394336db0564b3bca712d43042486dacc1 Mon Sep 17 00:00:00 2001
From: Lennart Poettering <lennart@poettering.net>
Date: Mon, 19 Dec 2011 17:40:31 +0100
Subject: [PATCH] man: extend sd-login(7) in regards to mixing D-Bus and
 synchronous library calls a bit (cherry picked from commit
 5079a105e701f17439635e76d8cb3052badbb34c)

---
 man/sd-login.xml |   12 +++++++++++-
 1 files changed, 11 insertions(+), 1 deletions(-)

diff --git a/man/sd-login.xml b/man/sd-login.xml
index 9926d2b..3fc0e16 100644
--- a/man/sd-login.xml
+++ b/man/sd-login.xml
@@ -77,13 +77,23 @@
                 you need to utilize the D-Bus API of
                 systemd-logind, instead.</para>
 
-                <para>These functions access data in
+                <para>These functions synchronously access data in
                 <filename>/proc</filename>,
                 <filename>/sys/fs/cgroup</filename> and
                 <filename>/run</filename>. All of these are virtual
                 file systems, hence the runtime cost of the accesses
                 is relatively cheap.</para>
 
+                <para>It is possible (and often a very good choice) to
+                mix calls to the synchronous interface of
+                <filename>sd-login.h</filename> with the asynchronous
+                D-Bus interface of systemd-logind. However, if this is
+                done you need to think a bit about possible races
+                since the stream of events from D-Bus and from
+                <filename>sd-login.h</filename> interfaces such as the
+                login monitor are asynchronous and not ordered against
+                each other.</para>
+
                 <para>If the functions return string arrays, these are
                 generally NULL terminated and need to be freed by the
                 caller with the libc
